packages/flutter/lib/src/widgets/restoration.dart

```diff
@@ -0,0 +1,1031 @@
+///    restoration data of a [StatefulWidget] instead of manually interacting
+///    with [RestorationScope]s and [RestorationBucket]s.
+///  * [RestorationManager], which describes the basic concepts of state
+///    restoration in Flutter.
+class UnmanagedRestorationScope extends InheritedWidget {
+  /// Creates an [UnmanagedRestorationScope].
+  const UnmanagedRestorationScope({
+    Key key,
+    this.bucket,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  /// The [RestorationBucket] that this widget will insert into the widget tree.
+  ///
+  /// Descendant widgets may obtain this bucket via [RestorationScope.of].
+  final RestorationBucket bucket;
+
+  @override
+  bool updateShouldNotify(UnmanagedRestorationScope oldWidget) {
+    return oldWidget.bucket != bucket;
+  }
+}
+
+/// Inserts a child bucket of [RestorationManager.rootBucket] into the widget
+/// tree and makes it available to descendants via [RestorationScope.of].
+///
+/// This widget is usually used close to the root of the widget tree to enable
+/// the state restoration functionality for the application. For all other use
+/// cases, consider using a regular [RestorationScope] instead.
+///
+/// {@macro flutter.widgets.restoration.scope}
+///
+/// The exact behavior of this widget depends on its ancestors: When the
+/// [RootRestorationScope] does not find an ancestor restoration bucket via
+/// [RestorationScope.of] it will claim a child bucket from the root restoration
+/// bucket ([RestorationManager.rootBucket]) using the provided [restorationId]
+/// and inserts that bucket into the widget tree where descendants may access it
+/// via [RestorationScope.of]. If the [RootRestorationScope] finds a non-null
+/// ancestor restoration bucket via [RestorationScope.of] it will behave like a
+/// regular [RestorationScope] instead: It will claim a child bucket from that
+/// ancestor and insert that child into the widget tree.
+///
+/// Unlike the [RestorationScope] widget, the [RootRestorationScope] will
+/// grantee that descendants have a bucket available for storing restoration
+/// data as long as [restorationId] is not null and [RestorationManager] is
+/// able to provide a root bucket. In other words, it will force-enable
+/// state restoration for the subtree.
+///
+/// If [restorationId] is null, no bucket is made available to descendants,
+/// which effectively turns of state restoration for this subtree.
+///
+/// The root restoration bucket can only be retrieved asynchronously from the
+/// [RestorationManager]. To ensure that the provided [child] has its
+/// restoration data available the first time it builds, the
+/// [RootRestorationScope] will build an empty [Container] instead of the actual
+/// [child] until the root bucket is available. To hide the empty container from
+/// the eyes of users, the [RootRestorationScope] also delays rendering the
+/// first frame while the container is shown. On most platforms, this keeps the
```

Comment:
Are there any platforms that this doesn't work on? Or do you just mean platforms that don't use splash screens (e.g. Windows, Linux)?

Reply:
Yeah, that's what I meant. Changed the comment to indicate that better.